crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -288,7 +291,229 @@ impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::CompleteAuthorizeData
+                types::PaymentsResponseData::TransactionResponse {
+                    resource_id,
+                    redirection_data,
+                    ..
+                } => {
+                    let connector_transaction_id = match resource_id {
+                        types::ResponseId::NoResponseId => None,
+                        types::ResponseId::ConnectorTransactionId(id)
+                        | types::ResponseId::EncodedData(id) => Some(id),
+                    };
+
+                    let encoded_data = payment_data.connector_response.encoded_data.clone();
+                    let connector_name = router_data.connector.clone();
+
+                    let authentication_data = redirection_data
+                        .map(|data| utils::Encode::<RedirectForm>::encode_to_value(&data))
+                        .transpose()
+                        .change_context(errors::ApiErrorResponse::InternalServerError)
+                        .attach_printable("Could not parse the connector response")?;
+
+                    if router_data.status == enums::AttemptStatus::Charged {
+                        metrics::SUCCESSFUL_PAYMENT.add(&metrics::CONTEXT, 1, &[]);
+                    }
+
+                    let capture_update = storage::CaptureUpdate::ResponseUpdate {
+                        status: Some(capture_status),
+                        connector_transaction_id: connector_transaction_id.clone(),
+                        error_code: None,
+                        error_message: None,
+                        error_reason: None,
+                    };
+                    let amount_captured = router_data
+                        .amount_captured
+                        .get_required_value("capture_amount")?;
+                    let previously_captured_amount =
+                        payment_data.payment_intent.amount_captured.unwrap_or(0);
+                    let authorized_amount = payment_data.payment_attempt.amount;
+                    let total_amount_captured = previously_captured_amount + amount_captured;
+
+                    let attempt_status = match capture_status {
+                        enums::CaptureStatus::Charged => {
+                            match authorized_amount.cmp(&total_amount_captured) {
+                                Ordering::Greater => enums::AttemptStatus::PartialCharged,
+                                Ordering::Equal => enums::AttemptStatus::Charged,
+                                Ordering::Less => {
+                                    Err(errors::ApiErrorResponse::InvalidDataValue {
+                                        field_name: "capture_amount", //ideally should not come till here
+                                    })?
+                                }
+                            }
+                        }
+                        enums::CaptureStatus::Pending => enums::AttemptStatus::CaptureInitiated,
```

Comment:
This status should be for the first partial capture request initiated right, what if we get this status for upcoming captures? Let's consider this scenario
- First partial capture is succeeded. AttemptStatus -> PartialCharged,
- Second partial capture is Pending. AttemptStatus -> CaptureInitiated,
Now should the Attempt status be partial captured or should it be updated to CaptureInitiated?

cc: @jarnura

Reply:
My idea was like, only if previous capture succeeds or fails, we can do next capture. So attempt_status will be an indicator of last capture is success or failure or pending.

If first partial capture status is :

> failed: attempt_status = Won't change (Authorized)
> success: attempt_status = PartialCharged
> pending: attempt_status = CaptureInitiated


If subsequent partial capture status is :
(subsequent capture should fail if attempt_status is CaptureInitiated)

> failed: attempt_status = Won't change (Authorized/ PartialCharged)
> success: attempt_status = PartialCharged
> pending: attempt_status = CaptureInitiated

Reply:
When can we receive the status as pending during partial capture?